frame/bags-list/src/lib.rs

```diff
@@ -195,6 +210,28 @@ pub mod pallet {
 			let _ = Pallet::<T>::do_rebag(&dislocated, current_weight);
 			Ok(())
 		}
+
+		/// Move `heavier` directly in front of `lighter`.
```

Comment:
Given

```
heavy -> ... -> ... -> light
```

I as a 3rd person can pay to make this

```
... -> ... -> heavy -> light
```

which is unfair to `heavy` node. 

given this I think this should not be permissionless, although it would be a pity. WDYT? the scenario above is def uncool.

Reply:
I need to think about this more - but one approach would be to heavily restrict usage and say heavy can only go in front of light if light is the head. Significantly less helpful since once a node with a weight equal to the bag threhsold is at the head we can no longer do anything

Do you see a compelling use case for having this as a permissioned extrinsic?

Reply:
> Significantly less helpful since once a node with a weight equal to the bag threhsold is at the head we can no longer do anything

yeah, not a fan. 

> In the context of Polkadot I don't really see a justification for adding this as a permissioned extrinsic

I def. do think there's a use case for the permissioned version as well. A user within a bag is incentivized to call this _themselves_ if they have a better position in the bag.

Reply:
Stating the obvious, but the fundamental issue here is we don't know relative position of any non-terminal nodes - not sure if there is any way of getting around that without adding more data to either the nodes or some storage item that keeps track of node index - both of which sound like too much added complexity at this stage

Reply:
> I def. do think there's a use case for the permissioned version as well. A user within a bag is incentivized to call this themselves if they have a better position in the bag.

I just mean if we make this extrinsic require root origin, will it ever get used? Like what is the case when governance would want to fiddle with the ordering within a single bag?

Reply:
I agree it should be permissioned and probably only callable by the heavy node

Reply:
Since this is going to be permissioned it will mean it is only callable by the stash address - so we should make sure this is callable through at least some proxy, or maybe work in a way to call it via the controller.

Reply:
Sounds better than having it permissionless.